types/hubot/index.d.ts

```diff
@@ -107,17 +109,23 @@ declare namespace Hubot {
     }
 
     interface Envelope {
+        id: string;
         room: string;
         user: User;
         message: Message;
     }
 
-    class Response<A extends Adapter = Adapter, M extends Message = Message> {
-        match: RegExpMatchArray;
-        message: Message;
+    class Response<
+        A extends Adapter = Adapter,
+        M extends Message = Message,
+        R extends RegExpMatchArray | { [key: string]: string } = RegExpMatchArray
```

Comment:
for this change the matche can be a RegExpMatchArray *OR* a custom object

https://github.com/hubotio/hubot/blob/main/src/Robot.mjs#L77-L91